[DM-6390] Create a testing distrib package

jmatt opened the pull request:
Create a smaller, faster to build distrib package for use with testing tools. Modeled after lsst_distrib and lsst_apps.

```
new file:   .gitignore
new file:   SConstruct
new file:   ups/deploy_testing_distrib.build
new file:   ups/deploy_testing_distrib.cfg
new file:   ups/deploy_testing_distrib.table
```

jmatt committed:
Initial implementing of a deploy testing distrib.

	new file:   .gitignore
	new file:   SConstruct
	new file:   ups/deploy_testing_distrib.build
	new file:   ups/deploy_testing_distrib.cfg
	new file:   ups/deploy_testing_distrib.table

jhoblitt commented on ups/deploy_testing_distrib.build:
```diff
@@ -0,0 +1,2 @@
+@LSST BUILD@ &&
+build_lsst @PRODUCT@ @VERSION@ @REPOVERSION@
```
This shouldn't be needed.  I looked at a few of the existing "metapackages" and most of them do have it but at least `qserv_distrib` does not.

jhoblitt commented:
As an FYI (it doesn't affect anything), the DM branch naming convention is `tickets/DM-XXXX`.

jhoblitt commented on SConstruct:
```diff
@@ -0,0 +1,3 @@
+# -*- python -*-
+from lsst.sconsUtils import scripts
+scripts.BasicSConstruct("deploy_testing_distrib")
```
If you change this to `scripts.BasicSConstruct("deploy_testing_distrib", noCfgFile=True)` the `deploy_testing_distrib.cfg` file can be removed.  The other metapackages I checked all have a `.cfg` file but it should be unnecessary.  As a sanity check, I confirmed that this change is functional. See `sconsUtils` for more info: https://github.com/lsst/sconsUtils/blob/763ede5c87e95b8fa030e1a295e433046841a622/python/lsst/sconsUtils/scripts.py#L59-L62

We should probably do this for all of the metapackages.

timj replied:
Yes, the `.build` files are there to teach `sconsUtils` how to link against this package. That's obviously not necessary for metapackages.

jhoblitt commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+.scon*
+config.log
+python/lsst/lsst/*/version.py
```
This path is bad.

timj commented on ups/deploy_testing_distrib.cfg:
```diff
@@ -0,0 +1,13 @@
+    libs=[],
+    hasDoxygenInclude=False,
+    hasSwigFiles=False,
+    )
```
(Moving my comment). `cfg` files are not needed for metapackages as they exist to teach `sconsUtils` how to link against C++ libraries).

jhoblitt replied:
@timj I opened a backlog story to fixup packages which don't need a `.cfg`.  https://jira.lsstcorp.org/browse/DM-6393

jhoblitt commented on ups/deploy_testing_distrib.table:
```diff
@@ -0,0 +1,4 @@
+setupRequired(meas_deblender)
+setupRequired(obs_test)
+setupOptional(obs_cfht)
+setupOptional(obs_subru)
```
We haven't been doing this for "metapackages" but technically `scons` and `sconsUtils` are deps.

jhoblitt replied:
All of the packages listed will pull in `afw`.  There's nothing wrong with that but it will result in a fairly long minimum build time.

jmatt replied:
Yeah I saw that. You have a better recommendation? I think the MVP should at least use the compiler and not be a external dependency.

jhoblitt replied:
They are minimal but `base` and `utils` both have some cpp in them.

jmatt committed:
Address comments in review.

  * Remove .build and .cfg files.
  * Remove extraneous lines in .gitignore.
  * Change SConstruct so no .cfg is required.

	modified:   .gitignore
	modified:   SConstruct
	deleted:    ups/deploy_testing_distrib.build
	deleted:    ups/deploy_testing_distrib.cfg

jmatt committed:
Change dependencies based on @jhoblitt's comment.

	modified:   ups/deploy_testing_distrib.table
